docs/infrastructure/TIMEOUTS.md

```diff
@@ -0,0 +1,59 @@
+# Timeouts
+
+Many load balancers have 60s configured as default timeout. Our API timeouts are designed to work within these bounds.
+
+## Preface: How network reaches Rivet
+
+```
+Client -> Cloudflare -> ELB -> Traefik -> api-monolith
+```
+
+## Infra timeouts
+
+-   Cloudflare: 100s ([source](https://developers.cloudflare.com/support/troubleshooting/cloudflare-errors/troubleshooting-cloudflare-5xx-errors/#error-524-a-timeout-occurred))
+    -   **Behavior** Returns a 524
+    -   Cannot be configured unless paying for Cloudflare Enterprise
+-	AWS NAT Gateway: 350 seconds idle (without keepalive) ([source](https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-troubleshooting.html#nat-gateway-troubleshooting-timeout))
+	-	**Behavior** Connection drop
+-	AWS ELB: 60 seconds ([source](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/config-idle-timeout.html#config-idle-timeout-console))
+	-	**Behavior** Connection drop
+-   Traefik: 60s ([source](https://github.com/rivet-gg/rivet/blob/c63067ce6e81f97b435e424e576fbd922b14f748/infra/tf/k8s_infra/traefik.tf#L65))
+    -   **Motivation** `api-helper` should always handle this error if everything is functioning correctly. This is meant to be less than Cloudflare to be able to show a Traefik-specific response.
+
+## API Timeouts
+
+We use long polling (i.e. `watch_index`) to implement real time functionality. This means we need to be cautious about existing timeouts.
+
+Current timeouts:
+
+-   Cloudflare: 100s ([source](https://developers.cloudflare.com/support/troubleshooting/cloudflare-errors/troubleshooting-cloudflare-5xx-errors/#error-524-a-timeout-occurred))
+    -   **Behavior** Returns a 524
+    -   Cannot be configured unless paying for Cloudflare Enterprise
+-   Traefik: 60s ([source](https://github.com/rivet-gg/rivet/blob/c63067ce6e81f97b435e424e576fbd922b14f748/infra/tf/k8s_infra/traefik.tf#L65))
+    -   **Motivation** `api-helper` should always handle this error if everything is functioning correctly. This is meant to be less than Cloudflare to be able to show a Traefik-specific response.
```

Comment:
Is this meant to be the same text as the above block?

Reply:
Clarified the sections